include/pybind11/pybind11.h

```diff
@@ -1075,15 +1083,27 @@ class class_ : public detail::generic_type {
     }
 
     /// Uses return_value_policy::reference_internal by default
+    template <typename Getter, typename Setter, typename... Extra>
+    class_ &def_property(const char *name, const Getter &fget, const Setter &fset, const Extra& ...extra) {
+        return def_property(name,
+                cpp_function(fget, is_method<type>(*this)), cpp_function(fset, is_method<type>(*this)),
+                return_value_policy::reference_internal, extra...);
+    }
```

Comment:
I'm not seeing how: if either of the `cpp_function, Setter` or `Getter, cpp_function` versions aren't there, then a call with one `cpp_function` and one lambda is going to end up at the fully-generic `Getter, Setter` template, which doesn't handle `cpp_function` arguments.